src/fides/api/models/privacy_experience.py

```diff
@@ -510,58 +509,57 @@ def new_experience_needed(
                 data={
                     "region": region,
                     "component": ComponentType.overlay,
-                    "delivery_mechanism": DeliveryMechanism.banner,  # Making this banner by default, just to be consistent.
                 },
             )
             added_experiences.append(overlay_experience)
 
-    return added_experiences, updated_experiences
+    return added_experiences
 
 
-def config_incompatible_with_region(
+def remove_config_from_matched_experiences(
     db: Session,
-    component_type: ComponentType,
-    delivery_mechanism: DeliveryMechanism,
-    region: PrivacyNoticeRegion,
-) -> bool:
-    """Returns True if an ExperienceConfig/PrivacyExperience is invalid for the given region
-
-    Our current primary check is that opt-in and notice only notices have to be displayed in
-    an overlay delivered by a banner.
-    """
-    return (
-        component_type == ComponentType.overlay
-        and delivery_mechanism == DeliveryMechanism.link
-        and bool(
-            get_privacy_notices_by_region_and_component(
-                db, region, ComponentType.overlay  # type: ignore[arg-type]
-            )
-            .filter(
-                PrivacyNotice.consent_mechanism.in_(
-                    [ConsentMechanism.opt_in, ConsentMechanism.notice_only]
-                )
-            )
-            .first()
-        )
-    )
+    experience_config: PrivacyExperienceConfig,
+    regions_to_unlink: List[PrivacyNoticeRegion],
+) -> List[PrivacyNoticeRegion]:
+    """Remove the config from linked PrivacyExperiences"""
+    if not regions_to_unlink:
+        return []
+
+    experiences_to_unlink: List[PrivacyExperience] = experience_config.experiences.filter(  # type: ignore[call-arg]
+        PrivacyExperience.region.in_(regions_to_unlink)
+    ).all()
+
+    for experience in experiences_to_unlink:
+        experience.unlink_experience_config(db)
+    return [experience.region for experience in experiences_to_unlink]  # type: ignore[misc]
 
 
 def upsert_privacy_experiences_after_config_update(
```

Comment:
yet another comment that i'm not suggesting we address in this PR 😄

as i wrap my head around all of this, part of me is starting to feel like the `PrivacyExperience` db model may be a bit superfluous? again, it makes total sense to have this as an API-level model - i like how simple it is as an API consumer to get this simple representation of experiences, all of which have been calculated by some relatively complex logic/maintenance on the BE. but do we get a lot of benefit from keeping this maintained in a db table?

i get the need to audit things and maintain a detailed history, so that could be one motivation -- but i still feel like the audit history could be pieced together by just the `PrivacyNoticeHistory` and `PrivacyExperienceConfigHistory` tables? we'd maybe need to provide some "utility" endpoints to make it easier on the consumer to piece that all together....

i guess what i get worried about here is the maintaining duplicate state across the `PrivacyExperience` and `PrivacyExperienceConfig` _and_ their associated history tables. i went down this line of thinking because it struck me as a bit funny that we're keeping a list of regions on the `PrivacyExperienceConfig` while _also_ maintaining links from `PrivacyExperience` records, whose main purpose is just to record the experience type for a given region (admittedly, that's a bit of an oversimplification). the method you've got here is pretty easy to follow, but we could potentially remove the need for it (and the `upsert_privacy_experiences_after_notice_update` function) and just rely on runtime lookups across the `PrivacyExperienceConfig` and `PrivacyNotice` tables to populate a `PrivacyExperience` API object that looks identical to what you have now - without maintaining a DB table (and associated history table).

i haven't totally thought this through! i can definitely see many good reasons to keep this in place. but it's a line of thinking i wanted to at least entertain as i am making my way through this. let me know what you think!

Reply:
To clarify one item up front, regions are _not_ duplicated across the `PrivacyExperienceConfig` and the `PrivacyExperience` models.  PrivacyExperience has a FK to PrivacyExperienceConfig but there is no corresponding list of regions on `PrivacyExperienceConfig`.   There is a `PrivacyExperienceConfig.regions` property but it's calculated from the PrivacyExperience FK relationship.

Reply:
> To clarify one item up front, regions are _not_ duplicated across the `PrivacyExperienceConfig` and the `PrivacyExperience` models. PrivacyExperience has a FK to PrivacyExperienceConfig but there is no corresponding list of regions on `PrivacyExperienceConfig`. There is a `PrivacyExperienceConfig.regions` property but it's calculated from the PrivacyExperience FK relationship.

aha, i think i missed this detail since that particular model attribute wasn't being changed in this PR - but it is an important detail that does settle the main source of my qualms, so thank you for recognizing my confusion and pointing this out!

> One of the main purposes of the Privacy Experience table at this point, in my mind, is to enforce that we only have one experience type per region at all times down to the database level.

this makes sense, and especially with the context you've provided on the logical nightmare that existed before (some of which is outside of this PR/discussion). i do think that at least some of the problems you were running into before are probably no longer as pertinent given the delivery mechanism removal you've got in this PR -- but i can certainly better see the motivation to have this sort of join/constraint table, given that changes from "both sides" (i.e. the `PrivacyNotice` and the `PrivacyExperienceConfig`) could prompt an update to what experience is shown for a given reason.

i do think that moving forward we need to be careful to _not_ duplicate data across the tables (i.e. `PrivacyNotice`, `PrivacyExperience` and `PrivacyExperienceConfig`). you likely have a much clearer picture on all of this than i do (still getting up to speed!), but given how interrelated these concepts are, i could imagine there being some confusion about what attributes belong where, as the data model evolves - so we'll need to take care when making updates (as you've done here!).

all that being said, i've been convinced that splitting the tables is a sensible and well-motivated choice. thank you for continuing to provide the context behind all of this!